### PR TITLE
Fixing CosmosSerializer comments

### LINKED ISSUE #771: Misleading documentation comment in CosmosSerializer.FromStream
According to the documentation comment on `CosmosSerializer.FromStream<T>` the caller will "ensure the stream is disposed":
https://github.com/Azure/azure-cosmos-dotnet-v3/blob/272aa1e80b0fd0bc7337332b9e355c151df21606/Microsoft.Azure.Cosmos/src/Serializer/CosmosSerializer.cs#L16-L17

This is misleading. To me, "ensure it's disposed" means "dispose it if it isn't already", but the reality is that it checks that the stream is disposed and throw an exception if not. The wording could be made clearer.

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

Updating the CosmosSerializer comments to make it clearer on who owns disposing of the stream.

closes #771

### CODE REVIEW

**Comment by j82w** [comment]
@bartelink and @thomaslevesque please take a look

**Comment by bartelink** [comment]
I prefer Thomas's wording as it does not suggest you only need to Dispose it just for the exception case - you always need to dispose it. (and uppercase the D in dispose)

**Comment by bartelink** [comment]
i.e.  https://github.com/Azure/azure-cosmos-dotnet-v3/issues/771#issuecomment-529213415

Waaaait, I scanned too fast, that is his wording ....

My problem with putting it this way is that one doesn't want to suggest that the Disposing is only relevant in the face of exceptions.

Thus I'd write:

>Implementation is responsible for Disposing of the stream. Failure to do so in all cases (including Exception cases) can result in a leak.

If that's not the actual semantic, I'd vary the wording ... is it ?

**Comment by thomaslevesque** [comment]
@bartelink how about this?

        /// The implementation is responsible for disposing of the stream,
        /// including when an exception is thrown, to avoid memory leaks
        /// if an exception is thrown.

**Comment by thomaslevesque** [comment]
As a side note, is there a reason why the caller doesn't dispose the stream itself? @j82w

**Comment by bartelink** [comment]
@thomaslevesque only change I'd make is `disposing` -> `Disposing`

**Comment by j82w** [comment]
Most serializers dispose of the stream when they do the conversion so it seemed redundant to put it in a wrapper to always dispose of it.

File: Microsoft.Azure.Cosmos/src/Serializer/CosmosSerializer.cs [hunk 1]
```diff
@@ -13,8 +13,8 @@ public abstract class CosmosSerializer
     {
         /// <summary>
         /// Convert a Stream of JSON to an object. 
-        /// The caller will ensure the stream is disposed. 
-        /// This will prevent memory leaks if an exception is hit.
+        /// The implementation is responsible for Disposing of the stream,
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Serializer/CosmosSerializer.cs]: Can SDK error out in-case stream is not disposed?

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Serializer/CosmosSerializer.cs]: It does error out.

**Pull request merged**